mitodl/mitxpro
# bundle optimization in webpack

#### Pre-Flight checklist

- [x] Screenshots for any changes that affect layout or styling
  

#### What are the relevant tickets?
#2339 

The changes in this PR are inspired by details mentioned in #2339 
#### What's this PR do?
- Generates a common bundle to be served for node_modules
- Removes packages included in non-test js bundles e.g. `chai`

#### How should this be manually tested?
- You will need to switch b/w `master and this branch` for testing.
- Generate the Webpack bundles on `master branch` (_How?_ Mentioned below) and note the bundle size
- Generate the webpack bundles on this  branch (_How?_ Mentioned below) and note the bundle size again

You should see the bundle sizing being reduced and a new bundle named `common` has been introduced in this new branch.

**How to generate the bundles?**
1- If you do `docker-comose up` your `watch` container would automatically generate the webpack bundles for you.
2- Get to watch container's shell and manually run `node node_modules/webpack/bin/webpack.js --config webpack.config.prod.js --bail` This will generate the prod bundles for you.
3- Use `webpack-bundle-analyer` to analyze the stats if you'd like. (Details mentioned in the associated ticket)

#### Where should the reviewer start?
 - Webpack bundle generation

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
**Before:**
![webpack-analyzer-ststs-master](https://user-images.githubusercontent.com/34372316/152948749-ddaeb55c-89d1-4e49-a725-34d7dff3c26a.png)

**After:**
![bundle_stats_after_rebase](https://user-images.githubusercontent.com/34372316/154235644-70bba7b7-9420-4f42-b1a6-487f7a5fa5ab.png)
